core/app/models/spree/gift_card.rb

```diff
@@ -0,0 +1,170 @@
+module Spree
+  class GiftCard < Spree.base_class
+    extend DisplayMoney
+    include Spree::SingleStoreResource
+    include Spree::Security::GiftCards if defined?(Spree::Security::GiftCards)
+
+    enum :state, { active: 0, redeemed: 1, canceled: 2, redeemed_by_order: 3, partialy_redeemed: 4 }
```

Comment:
@brozek95 let's not use enums for that, standard `state` string and state machine, same as we do for other models